packages/shared/src/reparent-children.ts

```diff
@@ -25,13 +31,15 @@ export const reparentChildren = (
         placeholderItems.push(placeholderItem);
 
         const srcElement = srcElements[index];
+        slotNames.push(srcElement.slot);
+        srcElement.removeAttribute('slot');
         const parentElement =
             srcElement.parentElement || srcElement.getRootNode();
         parentElement.replaceChild(placeholderItem, srcElement);
         newParent.append(srcElement);
     }
 
-    return function () {
-        restoreChildren(placeholderItems, srcElements);
+    return function (): Element[] {
+        return restoreChildren(placeholderItems, srcElements, slotNames);
```

Comment:
Use return so that consumers that want to do something more with the elements after their return can.